google/resource-snippets/pubsub-v1/alternatives/tf/pubsub.tf

```diff
@@ -0,0 +1,64 @@
+provider "google" {
+  project = "{PROJECT_ID}"
+  region  = "{REGION}"
+  zone    = "{ZONE}"
+}
+
+variable "deployment" {
+ default={}
+
+}
+
+variable "project_id" {
+ default={}
+
+}
+
+resource "google_pubsub_topic" "{TOPIC_REF}" {
+  name = "{TOPIC_NAME}"
+  labels= {
+    goog-dm=var.deployment
+  }
+}
+
+resource "google_pubsub_topic" "{BACKUP_TOPIC_REF}" {
+  name="{BACKUP_TOPIC_NAME}"
+  labels= {
+    goog-dm=var.deployment
+  }
+}
+
+resource "google_pubsub_subscription" "{SUBSCRIPTION_REF}" {
+  name = "{SUBSCRIPTION_NAME}"
+  labels= {
+    goog-dm=var.deployment
+  }
+  topic = google_pubsub_topic.{TOPIC_REF}.name
+  message_retention_duration="{MESSAGE_RETENTION_DURATION}"
+  retain_acked_messages={TRUE,FALSE}
+  ack_deadline_seconds = {ACK_DEADLINE_SECONDS}
+  dead_letter_policy {
+    dead_letter_topic = google_pubsub_topic.{BACKUP_TOPIC_REF}.id
+    max_delivery_attempts={MAX_DELIVERY_ATTEMPTS}
+
+  }
+  expiration_policy {
+    ttl="{TTL}"
+  }
+  retry_policy {
+    minimum_backoff="{MINIMUM_BACKOFF}"
+    maximum_backoff= "{MAXIMUM_BACKOFF}"
+
+  }
```

Comment:
Too much templating can actually make the template more difficult to use, because you're forced to provide values for fields you don't care about or you don't actually want to set. For the purpose of this repo, it's OK to only provide a handful of templated values just to showcase how it's done, and users can refer to the official documentation if they want to set more fields.

What do you think about creating 1 topic and 1 subscription for our examples, and remove the `deadLetterPolicy` and backup topic concepts? I think that'll make this example easier to understand. Same for the DM configs.